packages/twenty-front/src/modules/object-record/record-filter/utils/isRecordMatchingFilter.ts

```diff
@@ -0,0 +1,266 @@
+import { ObjectMetadataItem } from '@/object-metadata/types/ObjectMetadataItem';
+import {
+  AndObjectRecordFilter,
+  BooleanFilter,
+  DateFilter,
+  FloatFilter,
+  FullNameFilter,
+  LeafObjectRecordFilter,
+  NotObjectRecordFilter,
+  ObjectRecordFilter,
+  OrObjectRecordFilter,
+  StringFilter,
+  URLFilter,
+  UUIDFilter,
+} from '@/object-record/record-filter/types/ObjectRecordFilter';
+import { isMatchingBooleanFilter } from '@/object-record/record-filter/utils/isMatchingBooleanFilter';
+import { isMatchingDateFilter } from '@/object-record/record-filter/utils/isMatchingDateFilter';
+import { isMatchingFloatFilter } from '@/object-record/record-filter/utils/isMatchingFloatFilter';
+import { isMatchingStringFilter } from '@/object-record/record-filter/utils/isMatchingStringFilter';
+import { isMatchingUUIDFilter } from '@/object-record/record-filter/utils/isMatchingUUIDFilter';
+import { FieldMetadataType } from '~/generated-metadata/graphql';
+import { isAnObject } from '~/utils/isAnObject';
+import { isDefined } from '~/utils/isDefined';
+import { isEmptyObject } from '~/utils/isEmptyObject';
+
+export const isRecordMatchingFilter = ({
+  record,
+  filter,
+  objectMetadataItem,
+}: {
+  record: any;
+  filter: ObjectRecordFilter;
+  objectMetadataItem: ObjectMetadataItem;
+}) => {
+  if (Object.keys(filter).length === 0) {
+    return true;
+  }
+
+  const currentLevelFilterMatches: boolean[] = [];
+
+  // We consider all the keys at the same level as an "and"
+  for (const filterKey in filter) {
+    if (filterKey === 'and') {
+      const filterValue = (filter as AndObjectRecordFilter).and;
+
+      if (!Array.isArray(filterValue)) {
+        continue;
```

Comment:
should'nt we say that currentLevelFilterMatches.push(true) in this case too?

Reply:
And cannot have something else than an array, we should throw instead.